common/app/views/support/cleaner/AmpEmbedCleaner.scala

```diff
@@ -85,30 +92,86 @@ case class AmpEmbedCleaner(article: Article) extends HtmlCleaner {
     }
 
     def getTrackIdFromUrl(soundcloudUrl: String): Option[String] = {
-      val pattern = ".*soundcloud.com%2Ftracks%2F(\\d+).*".r
-      soundcloudUrl match {
-        case pattern(trackId) => Some(trackId)
+      val pattern = ".*api.soundcloud.com/tracks/(\\d+).*".r
+      URLDecoder.decode(soundcloudUrl,"UTF-8") match {
+        case pattern(trackId) => {
+          Some(trackId)}
         case _ => None
       }
     }
 
+    def getSoundCloudElement(document: Document, iframeElement: Element): Option[Element] = {
```

Comment:
This one is used in the clean function in ampAudioElements, but also in cleanAmpEmbed which is not part of ampAudioElements. I will move cleanAmpEmbed to the bottom of the file  as this method should be run after all the others and it adds to the confusion in its current location.